### PR TITLE
Fix MM-57406: prevent IPv6 hex segments from parsing as emoji

### LINKED ISSUE #26538: Parts of IPv6 addresses in messages parsed as emoticons
#### Summary
IPv6 addresses are parsed for emoticons, making the address hard to read.

#### Steps to reproduce
Add a custom emoticon with name that matches part of a IPv6-address, for example "beef".
Send a message with an IPv6-address with that part: 2001:18:1:beef::/64
IPv6 address displayed as 
image

#### Expected behavior
IPv6 addresses should be displayed as is, without parsing for emoticons.

#### Observed behavior (that appears unintentional) 
2001:db8:1:beef::/64 is displayed correctly but 2001:18:1:beef::/64 is not.

### PR DESCRIPTION
**Pull request by mgdelacroix**
#### Summary

The webapp's named-emoji regex (`EMOJI_PATTERN` in `webapp/channels/src/utils/emoticons.tsx`) had no boundary check on either side of the colons, so any colon-wrapped alphanumeric run was tokenized as an emoji. In IPv6 addresses like `2001:18:1:beef::/64`, both `:18:` and `:beef:` matched; `:beef:` rendered as a custom emoji image whenever one was defined.

This PR adds lookbehind and lookahead boundary checks so a `:name:` token only matches when neither side abuts an alphanumeric/underscore character. The new semantics match the server parser at `server/public/shared/markdown/emoji.go`, which already rejects this case via a prev-char guard and a trailing `\B` anchor.

Lookbehind is a regex *syntax* feature and is not lowered by `@babel/preset-env`, so this PR also bumps the webapp's babel `safari` target from `16.2` to `16.4`. The real minimum supported Safari is well above `16.4`, so the previous target was stale.

Behavior changes:
  - `2001:18:1:beef::/64`, `2001:cafe:1:dead::/64`, `100:100:100:100:100:100` — no longer tokenized.
  - `asdf:goat:asdf` — no longer tokenized (matches existing server behavior, e.g. `Thisisnotan:emoji:` at `server/public/shared/markdown/emoji_test.go:182`).
  - `:beef:`, `(:beef:)`, `:beef:.`, `:beef::dash:` — still tokenized.

QA steps:
1. Create a custom emoji named `beef`.
2. Post `2001:18:1:beef::/64` in a channel.
3. Expected: the message renders as literal text with no emoji image; previously the `:beef:` segment rendered as the emoji.
4. Confirm `:beef:` on its own still renders as the emoji.

#### Ticket Link

Fixes: https://mattermost.atlassian.net/browse/MM-57406
Fixes: https://github.com/mattermost/mattermost/issues/26538

#### Release Note

```release-note
Fixed IPv6 addresses containing hex segments (e.g. `:beef:`) being incorrectly rendered as custom emoji in the webapp.
```

## Change Impact: 🟢 Low

**Regression Risk:** The change is confined to the emoji-parsing regex in a webapp utility and a minor Babel target bump; added tests cover IPv6 and edge cases, minimizing regression risk.  
**QA Recommendation:** Standard QA — verify custom emoji rendering and that IPv6-like segments (e.g., `2001:18:1:beef::/64`, `asdf:goat:asdf`) remain literal while standalone/punctuation-wrapped `:name:` still render as emoji.

*Generated by CodeRabbitAI*

[Review Change Stack](https://app.coderabbit.ai/change-stack/mattermost/mattermost/pull/36541)

### CODE REVIEW

File: webapp/channels/src/utils/emoticons.tsx [hunk 1]
```diff
@@ -25,7 +25,7 @@ export const emoticonPatterns: { [key: string]: RegExp } = {
     broken_heart: /(^|\B)(\\?<\/3|\\?&lt;\/3)($|\b)/g, // </3
 };
 
-export const EMOJI_PATTERN = /(:([a-zA-Z0-9_+-]+):)/g;
+export const EMOJI_PATTERN = /(?<![A-Za-z0-9_])(:([a-zA-Z0-9_+-]+):)(?![A-Za-z0-9_])/g;
```
[the remark this answers is not in the document]
Comment (mgdelacroix) [reply webapp/channels/src/utils/emoticons.tsx]: All supported browser targets should work fine with negative lookbehind.

[the remark this answers is not in the document]
Comment (mgdelacroix) [reply webapp/channels/src/utils/emoticons.tsx]: Fixed in bc7af17b5c

**Review by lindy65 — APPROVED** [review]
Thanks @mgdelacroix - tested as per test steps on the Jira ticket and this is fixed/working as expected 👍

**Review by hmhealey — APPROVED** [review]
